types/trace/trace.go

```diff
@@ -7,8 +7,8 @@ import (
 	"strconv"
 )
 
-// TraceeEvent is a user facing data structure representing a single event
-type TraceeEvent struct {
+// EbpfEvent is a single result of an ebpf event process. It is used as a payload later delivered to tracee-rules.
+type EbpfEvent struct {
```

Comment:
Nit: Not sure what's the best name for the event payload generated by tracee-ebpf, but if we use acronyms in Go they're usually capital letters, i.e. `EBPFEvent`. At least that's what I've seen in Go SDK

Reply:
I'm not so sure what name is best Fine by me, I'll change it to `EBPFEvent`.
So since you've approved, anything else before we merge this?

Reply:
Why not just call it Event?

Reply:
> Why not just call it Event?

I initially suggested this in the PR but @AsafEitani and @danielpacak had their reservations about it. I sort of understand, in a talk I had with @AsafEitani we kept confusing between the two, however I do think the `trace` package solves that issue (`external` was much less meaningful)

Reply:
I initially had reservations about it because `external.Event` and `types.Event` were confusing to discuss in conversation.
Now that it is split to `detect` and `trace` in `types` - I have no problems with calling them both `Event`. In fact, that was the initial state of the PR until @AsafEitani posted his reservations about it in #1455.
If @danielpacak approves calling it `trace.Event` I have no problem changing it back.

Reply:
Let's call it trace.Event.

Reply:
I'm just wondering if `detect.Event` that represent generic event with any payload, mandatory and custom headers belongs to the `detect` package. It's something shared between tracing and detection capabilities.

Reply:
Maybe we could move `detect.Event` to `protocol.Event`?

Reply:
I've updated the PR to reflect the above suggestion (including moving back to `trace.Event`)